src/uhid/joypad_ps5.cpp

```diff
@@ -360,12 +361,16 @@ void PS5Joypad::set_on_rumble(const std::function<void(int, int)> &callback) {
   this->_state->on_rumble = callback;
 }
 
+/**
+ * For a rationale behind this, see: https://github.com/LizardByte/Sunshine/issues/3247#issuecomment-2428065349
+ */
 static __le16 to_le_signed(float original, float value) {
-  auto le = htole16(value);
-  if (original < 0) { // adjust sign bit
-    le |= (1 << 15);  // set the last bit (bit 15) to 1
+  if (value < SHRT_MIN) {
+    value = SHRT_MIN;
+  } else if (value > SHRT_MAX) {
+    value = SHRT_MAX;
   }
```

Comment:
```suggestion
  value = std::clamp(value, SHRT_MIN, SHRT_MAX);
```

Could you simplify to this?

Reply:
Kind of, you'll have to cast to `float` because `SHRT_MIN/MAX` are `unsigned int`.  
I agree though it looks cleaner using `clamp`, I'm going to commit that!